integration/system/uuid_test.go

```diff
@@ -0,0 +1,23 @@
+package system
+
+import (
+	"context"
+	"testing"
+
+	"github.com/docker/docker/api/types/versions"
+	"github.com/google/uuid"
+	"gotest.tools/assert"
+	"gotest.tools/skip"
+)
+
+func TestUUIDGeneration(t *testing.T) {
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.41"), "ID format changed")
+	defer setupTest(t)()
+
+	c := testEnv.APIClient()
+	info, err := c.Info(context.Background())
+	assert.NilError(t, err)
+
+	_, err = uuid.Parse(info.ID)
```

Comment:
Wondering to what extent we need to check the actual format, given https://github.com/moby/moby/blob/18b23067be82054a6384fc5934ecbb0276d1862b/api/swagger.yaml#L3365-L3370

Orthogonal to this; we should look at what purpose this field now serves. In a multi-node setup (Swarm cluster), there's also `Swarm.NodeID`, which is unique in the cluster, and is the ID that's used when running `docker node ls` (was even considering to copy that value here, but of course it wouldn't be there if swarm-mode is not active)

Reply:
@thaJeztah I'm not sure what to do with the swarmkit NodeID. I'd like to unblock this PR though. I personally don't need a UUID check, so I could just check for the non-emptiness of the ID. cc @justincormack